VL.Data/Data/ApplicationDbContextFactory.cs

```diff
@@ -0,0 +1,16 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Design;
+
+namespace VL.Shared.Data
+{
+    public class ApplicationDbContextFactory : IDesignTimeDbContextFactory<ApplicationDbContext>
+    {
+        public ApplicationDbContext CreateDbContext(string[] args)
+        {
+            var optionsBuilder = new DbContextOptionsBuilder<ApplicationDbContext>();
+            optionsBuilder.UseSqlServer("Server=(localdb)\\MSSQLLocalDB;Database=EcommerceDb;Trusted_Connection=True;MultipleActiveResultSets=true");
```

Comment:
This poses a problem here, how would you retrieve your connection string from the appsettings using the IOptions model

Reply:
Once the scaffolding is completed, which it is, this code serves no additional purpose. I can remove it.

Reply:
I would like to talk about the issue it creates if left in the code.

Reply:
IDesignTimeDbContextFactory is an interface provided by Microsoft.EntityFrameworkCore.Design. My understanding is it's sole purpose is to create a new DbContext instance in order to be available solely during design time in order to aid in things such as migrations or scaffolding. The connection string is hard coded into the options builder. Were you suggesting I could/should used the IOptions model in order to retrieve the connection string from appsettings? If so, I am not sure this would be available at design time, but I'm happy to try it.